app/src/main/java/com/coveros/coverosmobileapp/website/CustomWebViewClient.java

```diff
@@ -12,149 +12,143 @@
 import com.coveros.coverosmobileapp.blogpost.BlogPostReadActivity;
 import com.coveros.coverosmobileapp.blogpost.BlogPostsListActivity;
 
+import org.jsoup.nodes.Document;
+
 /**
  * Created by EPainter on 6/16/2017.
  * Provides Custom WebView Client that only loads Coveros-related content through WebView.
  * All externally related information is processed in a web browser
  */
 class CustomWebViewClient extends WebViewClient {
 
-    private boolean weAreConnected;
     private boolean isBlogPost;
-    private String savedClassName;
-    private int postID;
     private MainActivity mainActivity;
+    private int postId;
+    private String classNames;
+
     private static final String TAG = "CustomWebViewClient";
+    private static final String POST_ID_CLASS_PREFIX = "postid-";
   
-    CustomWebViewClient(MainActivity ma) {
-        mainActivity = ma;
-        weAreConnected = true;
+    CustomWebViewClient(MainActivity mainActivity) {
+        this.mainActivity = mainActivity;
     }
 
-    CustomWebViewClient() {
-        weAreConnected = true;
-    }
+    @Override
+    public boolean shouldOverrideUrlLoading(final WebView view, WebResourceRequest request) {
+        //If blog website or blog web page is categorically loaded (hybrid)
+        isBlogPost = false;
+        final String url = request.getUrl().toString();
 
-    MainActivity getMainActivity() {
-        return mainActivity;
-    }
+        DocumentGenerator urlHtml = new DocumentGenerator(url, new DocumentGenerator.DocumentGeneratorCallback() {
+            @Override
+            public void onDocumentReceived(Document document) {
+                classNames = document.body().className();
+                isBlogPost = checkIsBlogPost(classNames);
 
-    void setMainActivity(MainActivity ma) {
-        mainActivity = ma;
-    }
+                view.post(new RedirectManager(view, url));
 
-    boolean getConnection() {
-        return weAreConnected;
-    }
+            }
+        });
 
-    void setConnection(boolean answer) {
-        weAreConnected = answer;
-    }
+        // Create new thread to handle network operations
+        Thread getUrlHtmlThread = new Thread(urlHtml);
+        getUrlHtmlThread.start();
 
-    boolean getIsBlogPost() {
-        return isBlogPost;
-    }
+        return true;
 
-    void setIsBlogPost(Boolean blogPost) {
-        isBlogPost = blogPost;
     }
 
-    int getPostID() {
-        return postID;
+    public boolean checkIsBlogPost(String classNames){
```

Comment:
change this method body to just be `return classNames.contains(POST_ID_CLASS_PREFIX);`